main_test.go

```diff
@@ -46,32 +47,81 @@ func (t *MainTest) TestGetUserAgentWhenMetadataImageTypeEnvVarIsSet() {
 	os.Setenv("GCSFUSE_METADATA_IMAGE_TYPE", "DLVM")
 	defer os.Unsetenv("GCSFUSE_METADATA_IMAGE_TYPE")
 
-	userAgent := getUserAgent("AppName")
-	expectedUserAgent := strings.TrimSpace(fmt.Sprintf("gcsfuse/%s AppName (GPN:gcsfuse-DLVM)", getVersion()))
+	mountConfig := &config.MountConfig{}
+	userAgent := getUserAgent("AppName", getConfigForUserAgent(mountConfig))
+	expectedUserAgent := strings.TrimSpace(fmt.Sprintf("gcsfuse/%s AppName (GPN:gcsfuse-DLVM) (Cfg:0:0)", getVersion()))
 
 	ExpectEq(expectedUserAgent, userAgent)
 }
 
 func (t *MainTest) TestGetUserAgentWhenMetadataImageTypeEnvVarIsNotSet() {
-	userAgent := getUserAgent("AppName")
-	expectedUserAgent := strings.TrimSpace(fmt.Sprintf("gcsfuse/%s (GPN:gcsfuse-AppName)", getVersion()))
+	mountConfig := &config.MountConfig{}
+	userAgent := getUserAgent("AppName", getConfigForUserAgent(mountConfig))
+	expectedUserAgent := strings.TrimSpace(fmt.Sprintf("gcsfuse/%s (GPN:gcsfuse-AppName) (Cfg:0:0)", getVersion()))
 
 	ExpectEq(expectedUserAgent, userAgent)
 }
 
-func (t *MainTest) TestGetUserAgentWhenMetadataImageTypeEnvVarAndAppNameAreNotSet() {
-	userAgent := getUserAgent("")
-	expectedUserAgent := strings.TrimSpace(fmt.Sprintf("gcsfuse/%s (GPN:gcsfuse)", getVersion()))
+func (t *MainTest) TestGetUserAgentConfig() {
 
+	// Test Default Case - No File Cache
+	mountConfig := &config.MountConfig{}
+	userAgent := getUserAgent("AppName", getConfigForUserAgent(mountConfig))
+	expectedUserAgent := strings.TrimSpace(fmt.Sprintf("gcsfuse/%s (GPN:gcsfuse-AppName) (Cfg:0:0)", getVersion()))
 	ExpectEq(expectedUserAgent, userAgent)
+
+	// Test File Cache Enabled and Random Read Enabled
+	mountConfig1 := &config.MountConfig{
```

Comment:
would prefer separating the unit tests like TestGetUserAgentWhenFileCacheIsEnabled

Reply:
Done. Kindly review.